### PR TITLE
Structure SASS folder with ITCSS

### PR DESCRIPTION
**Pull request by samikeijonen**
## Changes proposed in this Pull Request:

- Simplify SASS folder structure using [ITCSS approach](https://www.xfive.co/blog/itcss-scalable-maintainable-css-architecture/). Conversation is happening in the issue #1215.
- The main goal is to have more readable structure.
- SASS folder structure should be easily maintained.
- Structure should be inline with current best practises. Meaning scalable, modular architecture.
- Structure should be good starting point in the bigger projects also.

## Related issue(s):

#1215.

### CODE REVIEW

**Comment by samikeijonen** [comment]
Folder structure should be able to read from `style.scss` file.

```css
/*--------------------------------------------------------------
>>> TABLE OF CONTENTS:
----------------------------------------------------------------
# Generic
	## Normalize
	## Box sizing
# Elements
# Layouts
	## Posts and pages
	## Comments
	## Widgets
# Vendors
	## Infinite scroll
# Components
	## Navigation
	## Media
		## Captions
		## Galleries
# Utilities
	## Accessibility
	## Alignments
	## Clearings
--------------------------------------------------------------*/

/*--------------------------------------------------------------
# Settings - Global variables, config switches.
--------------------------------------------------------------*/
@import "settings/variables";

/*--------------------------------------------------------------
# Tools - Default mixins and functions.
--------------------------------------------------------------*/
@import "tools/mixins";

/*--------------------------------------------------------------
# Generic - Ground-zero styles (resets, box-sizing, etc.).
--------------------------------------------------------------*/
/*--------------------------------------------------------------
## Normalize
--------------------------------------------------------------*/
@import "generic/normalize";

/*--------------------------------------------------------------
## Box sizing
--------------------------------------------------------------*/
@import "generic/box-sizing";

/*--------------------------------------------------------------
# Elements - Unclassed HTML elements.
--------------------------------------------------------------*/
@import "elements/typography/typography";
@import "elements/elements";

/*--------------------------------------------------------------
# Layouts - class-based undecorated design patterns.
--------------------------------------------------------------*/
// @import "layouts/content-sidebar";
// @import "layouts/sidebar-content";

/*--------------------------------------------------------------
## Posts and pages
--------------------------------------------------------------*/
@import "layouts/posts-and-pages";

/*--------------------------------------------------------------
## Comments
--------------------------------------------------------------*/
@import "layouts/comments";

/*--------------------------------------------------------------
## Widgets
--------------------------------------------------------------*/
@import "layouts/widgets";

/*--------------------------------------------------------------
## Vendors - Third-party plugins styles.
--------------------------------------------------------------*/
/*--------------------------------------------------------------
## Infinite scroll
--------------------------------------------------------------*/
@import "vendors/infinite-scroll";

/*--------------------------------------------------------------
# Components - specific UI components.
--------------------------------------------------------------*/
/*--------------------------------------------------------------
## Navigation
--------------------------------------------------------------*/
@import "components/navigation";

/*--------------------------------------------------------------
## Media
--------------------------------------------------------------*/
@import "components/media/media";

/*--------------------------------------------------------------
# Utilities - Helpers and overrides.
--------------------------------------------------------------*/
/*--------------------------------------------------------------
## Accessibility
--------------------------------------------------------------*/
@import "utilities/accessibility";

/*--------------------------------------------------------------
## Alignments
--------------------------------------------------------------*/
@import "utilities/alignments";

/*--------------------------------------------------------------
## Clearings
--------------------------------------------------------------*/
@import "utilities/clearings";
```

**Comment by samikeijonen** [comment]
Hmm having these comments are problematic because they get generated in `style.css` also. Any suggestions?

```css
/*--------------------------------------------------------------
# Settings - Global variables, config switches.
--------------------------------------------------------------*/
@import "settings/variables";

/*--------------------------------------------------------------
# Tools - Default mixins and functions.
--------------------------------------------------------------*/
```

**Comment by m-e-h** [comment]
I agree that the "vendors" folder can be confusing. I know it's sort of the standard name for 3rd party dependencies and such but ours isn't really a dependency folder. It's an optional folder for plugin styles. Would just calling it "plugins" be more indicative of what's inside?

@bahiirwa the reason normalize isn't residing in the same folder as the plugin styles is because it's specificity and purpose is completely different. You import normalize at the beginning and easily override it's styles if necessary while the plugins will likely contain styles that should be able to easily override others so it gets imported later.
I think that's kinda the context of this particular PR anyway.

**Comment by m-e-h** [comment]
PR **1234** :sparkles: That number can't happen too often.
I believe there's a prize if it gets merged. No pressure @samikeijonen :smiley:

**Comment by samikeijonen** [comment]
> Would just calling it "plugins" be more indicative of what's inside?

I was thinking about the same thing. At the moment I feel it should be named `plugins`.

**Comment by justnorris** [comment]
+1 for naming it plugins

File: sass/style.scss [hunk 1]
```diff
@@ -23,84 +23,108 @@ Nicolas Gallagher and Jonathan Neal https://necolas.github.io/normalize.css/
...
 /*--------------------------------------------------------------
-# Accessibility
+## Widgets
 --------------------------------------------------------------*/
-@import "modules/accessibility";
+@import "layouts/widgets";
 
 /*--------------------------------------------------------------
-# Alignments
+## Plugins - Third-party plugins styles.
```
Comment (Ismail-elkorchi) [inline sass/style.scss]: I think that this line should be # Plugins... instead of ## Plugins...

**Pull request closed without merging**